### DISCUSSION TITLE
Fetching and displaying json data

### DISCUSSION THREAD

**Discussion by djsedov** [vercel/next.js#58580] [Help] [answered]
### Summary

I'm using the app router and Strapi for storing data. When trying to access the api - console log gives the contents of an json file.
But when i'm trying do display the json contents or a certain property or attribute on a page the <main> div is empty.

What am i doing wrong?

app router is here

http://194.67.78.236:3000/testfetch

API response from strapi

http://194.67.78.236:1337/api/categories

### Additional information

```js
`async function getData() {
  const res = await fetch('http://194.67.78.236:1337/api/categories')
  // The return value is *not* serialized
  // You can return Date, Map, Set, etc.
 
  if (!res.ok) {
    // This will activate the closest `error.js` Error Boundary
    throw new Error('Failed to fetch data')
  }
 
  return res.json()
}
 
export default async function Page() {
  const data = await getData()
 
  return <main>{data.name}</main>
}
`
```

### Example

When i try this construction - json data shows in console, but still not rendering on a page

`
"use client";
import { useState, useEffect } from 'react'
 
function Profile() {
  const [data, setData] = useState(null)
  const [isLoading, setLoading] = useState(true)
 
  useEffect(() => {
    fetch('http://194.67.78.236:1337/api/categories')
      .then((res) => res.json())
      .then((data) => {
        setData(data)
        setLoading(false)
      })
  }, [])
 
  if (isLoading) return <p>Loading...</p>
  if (!data) return <p>No profile data</p>
 
  return (
    <ul>
      <h1>{data.id}</h1>
      <p>{data.name}</p>
    </ul>
  )
}

export default Profile;

`

**Comment by icyJoseph** [comment]
Inside Page, before the return, could you do `console.log(typeof data, typeof data.name)` and see what that prints?

**Reply by djsedov** [reply]
![изображение](https://github.com/vercel/next.js/assets/60842995/63e7b0ab-9296-4ac3-845a-192cc004e707)

**Reply by icyJoseph** [reply]
? 

This is a server component, the logs are on the server (terminal).

Also, I assume the DOM also does not have the contents? (Elements tab in the browser)

**Comment by icyJoseph** [comment, marked as answer]
@djsedov Seeing the data from strapi, I think the error is that the data is shaped like this:

```json
{
 "data": []
}
```

So, the first problem is that you have to do this:

```js
export default async function Page() {
  // getData gives you an object, with a nested key, named data
  const { data } = await getData()
 
  return <main>{data.name}</main>
}
```

The second problem is that, http://194.67.78.236:1337/api/categories, contains a list within data, so there's no `data.name` to access.


Try this:

```js
export default async function Page() {
  // back to how you had it before
  const obj = await getData()
 
  return <pre>{JSON.stringify(obj, null, 2)}</prev>
}
```

That'll help you see the problem. You need understand that you need to see into `obj.data`.

**Reply by djsedov** [reply]
Big thanks! Your andswer helped so much to understand and solve the problem :)
